launch/launch/actions/opaque_function.py

```diff
@@ -0,0 +1,72 @@
+# Copyright 2018 Open Source Robotics Foundation, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Module for the OpaqueFunction action."""
+
+import collections
+from typing import Any
+from typing import Callable
+from typing import Dict
+from typing import Iterable
+from typing import List
+from typing import Optional
+from typing import Text
+
+from ..action import Action
+from ..launch_context import LaunchContext
+from ..utilities import ensure_argument_type
+
+
+class OpaqueFunction(Action):
+    """
+    Action that executes a Python function.
+
+    The signature of the function should be:
+
+    .. code-block:: python
+
+        def function(
+            context: LaunchContext,
+            *args,
+            **kwargs
+        ) -> Optional[List[LaunchDescriptionEntity]]:
```

Comment:
Actually it can be Actions or events. I believe list of entities is more accurate. All entities are visited and actions which are visited are executed. So in this way actions can return non-action entities (there are none really right now other than `LaunchDescription`).

There's no technical issue here, but maybe some inconsistency in the use of the type annotations.